src/impl_methods.rs

```diff
@@ -2096,7 +2105,16 @@ where
         let view_stride = self.strides.axis(axis);
         // use the 0th subview as a map to each 1d array view extended from
         // the 0th element.
-        self.index_axis_mut(axis, 0).map_mut(|first_elt: &mut A| {
+        let mut first_subview = if view_len != 0 {
+            self.index_axis_mut(axis, 0)
+        } else {
+            let smaller_zero_dim = <D::Smaller as Dimension>::zeros(self.dim.ndim() - 1);
+            let ptr = std_ptr::NonNull::<A>::dangling().as_ptr();
+            unsafe {
```

Comment:
I'd like a comment here to explain why this is a safe thing to do (same in the other spot).

Reply:
@LukeMathWalker is it be enough to point to [this](https://github.com/rust-ndarray/ndarray/blob/master/src/lib.rs#L1108) section in the `ArrayBase` docstring?
> The only exceptions are if the array is empty or the element
> type is zero-sized. In these cases, `ptr` may be dangling, but it must
> still be safe to [`.offset()`] the pointer along the axes.

Reply:
I'd suggest to extract it in a function, so that we can reuse code across the two calling sites and keep a better eye on it. It's probably better to copy the relevant section over, to have it close to the code.

Reply:
Ready